tests/test_utils/tests.py

```diff
@@ -263,6 +263,41 @@ def make_configuration_query():
                 list(Car.objects.all())
 
 
+class MultiDBRouter:
+    def allow_relation(self, obj1, obj2, **hints):
+        return True
+
+    def db_for_read(self, model, **hints):
+        if model == Car:
+            return "other"
+        return "default"
+
+    db_for_write = db_for_read
+
+
+@override_settings(DATABASE_ROUTERS=[f"{__name__}.MultiDBRouter"])
+class AssertNumQueriesTestsMultiDB(TestCase):
+    databases = {"default", "other"}
+
+    def test_assert_num_queries_all(self):
+        with self.assertNumQueries(3, using="__all__"):
+            Car.objects.count()
+            Person.objects.count()
+            PossessedCar.objects.count()
+
+    def test_assert_num_queries_all_specific(self):
+        with self.assertNumQueries(3, using={"default", "other"}):
+            Car.objects.count()
+            Person.objects.count()
+            PossessedCar.objects.count()
+
+    def test_assert_num_queries_specific(self):
```

Comment:
Note to self: Maybe add a test that also tests

`with self.assertNumQueries(2, using={"default"}), self.assertNumQueries(1, using={"other"})` to prove this continues to work

Reply:
Done.